### DISCUSSION TITLE
How to add the server name into the completion menu?

### DISCUSSION THREAD

**Discussion by nyngwang** [hrsh7th/nvim-cmp#779] [Q&A] [answered]
As title, is it possible from `[LSP]` to `[LSP] - tsserver` or so.

**Comment by daephx** [comment]
I was just looking for the same thing!
Could someone please provide an answer on if this is possible?

Given that it's pretty likely to have multiple language servers attached at one time, something like this would be pretty helpful.

**Reply by scflode** [reply]
Would be great if this is possible. The `entry.source.name` only contains the type (buffer, nvim_lsp).

**Reply by fitrh** [reply]
It is possible, you can use `entry.source.source.client.name`

**Comment by fitrh** [comment, marked as answer]
You can use `entry.source.source.client.name`, something like this

```lua
cmp.setup({
    formatting = {
        format = function(entry, item)
            item.menu = ({
	        buffer = "[Buffer]",
	        nvim_lsp = "[LSP]",
            })[entry.source.name]
            
            if entry.source.source.client then
                item.menu = ("%s - %s"):format(item.menu, entry.source.source.client.name)
            end

            return item
        end
    },
})
```

**Reply by scflode** [reply]
Awesome. Thank you!

**Reply by fitrh** [reply]
Thank you also for your reply, hence, this long-standing question appeared in my notifications
